src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -767,6 +790,59 @@ class spell_rog_vanish : public SpellScript
     }
 };
 
+class spell_rog_glyph_of_backstab_triggered : public SpellScriptLoader
+{
+public:
+    spell_rog_glyph_of_backstab_triggered() : SpellScriptLoader("spell_rog_glyph_of_backstab_triggered") { }
+
+    class spell_rog_glyph_of_backstab_triggered_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_rog_glyph_of_backstab_triggered_SpellScript);
+
+        typedef spell_rog_rupture::spell_rog_rupture_AuraScript RuptureAuraScript;
+
+        void HandleScript(SpellEffIndex effIndex)
+        {
+            PreventHitDefaultEffect(effIndex);
+
+            Unit* caster = GetCaster();
+            // search our Rupture aura on target
+            if (AuraEffect* aurEff = GetHitUnit()->GetAuraEffect(SPELL_AURA_PERIODIC_DAMAGE, SPELLFAMILY_ROGUE, 0x00100000, 0, 0, caster->GetGUID()))
+            {
+                RuptureAuraScript* ruptureAuraScript = dynamic_cast<RuptureAuraScript*>(aurEff->GetBase()->GetScriptByName(RuptureScriptName));
+                if (!ruptureAuraScript)
+                    return;
+
+                uint32& bonusDuration = ruptureAuraScript->BonusDuration;
+
+                // already includes duration mod from Glyph of Rupture
+                uint32 countMin = aurEff->GetBase()->GetMaxDuration();
+                uint32 countMax = countMin - bonusDuration;
+                // this glyph
+                countMax += 6000;
+
+                if (countMin < countMax)
+                {
+                    bonusDuration += 2000;
```

Comment:
It is very controversial to modify a variable of another class by reference. It's just int. There will be no performance gain from such tricks.